library/src/amd_detail/hipblas.cpp

```diff
@@ -16718,17 +16742,19 @@ try
 {
     if(info == NULL)
         return HIPBLAS_STATUS_INVALID_VALUE;
+    else if(trans != HIPBLAS_OP_N && trans != HIPBLAS_OP_T)
+        *info = -1;
     else if(m < 0)
         *info = -2;
     else if(n < 0)
         *info = -3;
     else if(nrhs < 0)
         *info = -4;
-    else if(A == NULL)
+    else if(A == NULL && m * n)
         *info = -5;
     else if(lda < m)
         *info = -6;
-    else if(B == NULL)
+    else if(B == NULL && (m * nrhs || n * nrhs))
```

Comment:
How about `max(m, n) * nrhs`?

Reply:
I suppose that would work, I like this better though, it's more clear and doesn't depend on negatives getting handled earlier, it's also identical to some error checking code in rocSOLVER.
For now I just added comments in testing_gels.hpp (and (strided)_bathced) regarding the explicit values.